docs: improve, define Python library exception list (#16662)
Homebrew/brew: docs/Python-for-Formula-Authors.md

### Python declarations for applications

> Formulae for apps that require Python 3 **must** declare an unconditional dependency on `"python@3.x"`. These apps **must** work with the current Homebrew Python 3.x formula.

Reviewer: Should we use the same naming convention as other doc (i.e. `3.y`)? Or perhaps other way around and change other doc to `3.x`?

Reviewer 2: Agreed consistency would be nice, either way.

Author: Done

Remember: there are very limited cases for libraries (e.g. significant amounts of native code is compiled) so, if in doubt, do not package them.

> **We do not use the `python-` prefix for these kind of formulae!**

Reviewer 3: ```suggestion
**We do not use the `python-` prefix for these kinds of formulae!**
```

> ### Examples of allowed libaries in homebrew-core

Reviewer: ```suggestion
### Examples of allowed libraries in homebrew-core
```

* `numpy`, `scipy`: long build time, complex build process

* `cryptography`: builds with `rust`

> * `certifi`: patched formula to allow any Python-based formulae to leverage the brewed CA certs (see https://github.com/orgs/Homebrew/discussions/4691).

Reviewer 3: ```suggestion
* `certifi`: patched formula to allow any Python-based formulae to leverage the brewed CA certs (see <https://github.com/orgs/Homebrew/discussions/4691>).
```

Reviewer 3: Angle brackets are necessary to satisfy the MDL rules.

### `pip` vs. `setup.py`

> [PEP 453](https://legacy.python.org/dev/peps/pep-0453/#recommendations-for-downstream-distributors) makes a recommendation to downstream distributors (us) that sdist tarballs should be installed with `pip` instead of by invoking `setup.py` directly. For historical reasons we did not follow PEP 453, so some formulae still use `setup.py` installs. Nowadays, most of the formulae use `pip` as we have migrated them to this better way of installation.

Reviewer 3: ```suggestion
[PEP 453](https://legacy.python.org/dev/peps/pep-0453/#recommendations-for-downstream-distributors) makes a recommendation to downstream distributors (us) that sdist tarballs should be installed with `pip` instead of by invoking `setup.py` directly. For historical reasons we did not follow PEP 453, so some formulae still use `setup.py` installs. Nowadays, most of the formulae use `pip` as we have migrated them to this preferred method of installation.
```
